crates/cxx-qt-lib/src/gui/qimage.rs

```diff
@@ -0,0 +1,76 @@
+// SPDX-FileCopyrightText: 2023 Klarälvdalens Datakonsult AB, a KDAB Group company <info@kdab.com>
+// SPDX-FileContributor: Leon Matthes <leon.matthes@kdab.com>
+//
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+use crate::QByteArray;
+use cxx::{type_id, ExternType};
+use std::{
+    ffi::{c_char, CString},
+    mem::MaybeUninit,
+};
+
+#[cxx::bridge]
+mod ffi {
+
+    unsafe extern "C++" {
+        include!("cxx-qt-lib/qbytearray.h");
+        type QByteArray = crate::QByteArray;
+
+        include!("cxx-qt-lib/qimage.h");
+        type QImage = super::QImage;
+
+        #[rust_name = "is_null"]
+        fn isNull(self: &QImage) -> bool;
+    }
+
+    #[namespace = "rust::cxxqtlib1"]
+    unsafe extern "C++" {
+        include!("cxx-qt-lib/common.h");
+
+        #[doc(hidden)]
+        #[rust_name = "qimage_drop"]
+        fn drop(image: &mut QImage);
+
+        #[doc(hidden)]
+        #[rust_name = "qimage_init_from_data"]
+        unsafe fn qimageInitFromData(data: &QByteArray, format: *const c_char) -> QImage;
+    }
+}
+
+#[repr(C)]
+pub struct QImage {
```

Comment:
needs docs, but as discussed needs to stated it needs to be marked as experimental / WIP etc